File: app/src/main/java/com/android/learn/mpresenter/ProjectPresenter.java
package com.android.learn.mpresenter;


import com.android.base.mmodel.ProjectListData;
import com.android.base.mpresenter.BasePresenter;
import com.android.base.thirdframe.rxjava.BaseObserver;
import com.android.learn.mcontract.ProjectContract;

import io.reactivex.Observable;


public class ProjectPresenter extends BasePresenter<ProjectContract.View> implements ProjectContract.Presenter {

    private int mCurrentPage = 1;

    @Override
    public void getProjectInfo(int page, int cid) {
        Observable observable = mRestService.getProjectListData(page, cid);
        addSubscribe(observable, new BaseObserver<ProjectListData>() {
            @Override
            public void onNext(ProjectListData projectListData) {
                mView.showProjectList(projectListData, false);
            }
        });
    }

    @Override
    public void onRefreshMore(int cid) {
        Observable observable = mRestService.getProjectListData(-1, cid);
        addSubscribe(observable, new BaseObserver<ProjectListData>() {
            @Override
            public void onNext(ProjectListData projectListData) {
                mView.showProjectList(projectListData, true);
            }
        });
    }

    @Override
    public void onLoadMore(int cid) {
        ++mCurrentPage;
        Observable observable = mRestService.getProjectListData(mCurrentPage, cid);
        addSubscribe(observable, new BaseObserver<ProjectListData>() {
            @Override
            public void onNext(ProjectListData projectListData) {
                mView.showProjectList(projectListData, false);
            }
        });
    }


}

File: app/src/main/java/com/android/learn/mpresenter/ProjectPresenter.kt
package com.android.learn.mpresenter


import com.android.learn.base.mmodel.ProjectListData
import com.android.learn.base.mpresenter.BasePresenter
import com.android.learn.base.thirdframe.rxjava.BaseObserver
import com.android.learn.mcontract.ProjectContract

import io.reactivex.Observable


class ProjectPresenter : BasePresenter<ProjectContract.View>(), ProjectContract.Presenter {

    private var mCurrentPage = 1

    override fun getProjectInfo(page: Int, cid: Int) {
        val observable = mRestService.getProjectListData(page, cid)
        addSubscribe(observable, object : BaseObserver<ProjectListData>(true) {
            override fun onNext(projectListData: ProjectListData) {
                mView!!.showProjectList(projectListData, false)
            }
        })
    }

    override fun onRefreshMore(cid: Int) {
        val observable = mRestService.getProjectListData(-1, cid)
        addSubscribe(observable, object : BaseObserver<ProjectListData>(false) {
            override fun onNext(projectListData: ProjectListData) {
                mView!!.showProjectList(projectListData, true)
            }
        })
    }

    override fun onLoadMore(cid: Int) {
        ++mCurrentPage
        val observable = mRestService.getProjectListData(mCurrentPage, cid)
        addSubscribe(observable, object : BaseObserver<ProjectListData>(false) {
            override fun onNext(projectListData: ProjectListData) {
                mView!!.showProjectList(projectListData, false)
            }
        })
    }


}

